### PR TITLE
Fix quark fp8 format loading.

### PR DESCRIPTION
**Pull request by fxmarty-amd**
This PR fixes an issue in the order `requantize_with_max_scale` is called when loading a checkpoint using quark config.json/checkpoint format.

Compare the following:
- neuralmagic fp8: https://github.com/vllm-project/vllm/blob/7a8987dac5f0ed0c798a73e8b4ec8f5e640bc63a/vllm/model_executor/layers/quantization/fp8.py#L303-L317
- quark fp8: https://github.com/ROCm/vllm/blob/685281948d39d88f20f0290848b24d8b6139b93c/vllm/model_executor/layers/quantization/quark/schemes/quark_w8a8_fp8.py#L35-L48

`normalize_e4m3fn_to_e4m3fnuz` should be called before `requantize_with_max_scale`.

### CODE REVIEW

**Comment by gshtras** [comment]
A separate PR is not required. Once the upstream one is accepted, we'll get it within 1 week usually

**Comment by fxmarty-amd** [comment]
@gshtras I think ROCm/vllm is used more widespread internally so if this can get included here before merging upstream, it is nice, but I can close this if you prefer!

File: vllm/model_executor/layers/quantization/quark/schemes/quark_w8a8_fp8.py [hunk 1]
```diff
@@ -33,21 +33,21 @@ def process_weights_after_loading(self, layer) -> None:
...
+                    weight=layer.weight,
+                    weight_scale=layer.weight_scale,
                     input_scale=layer.input_scale)
                 if input_scale is not None:
                     layer.input_scale = Parameter(input_scale,
                                                   requires_grad=False)
 
+            max_w_scale, weight = requantize_with_max_scale(
+                weight=weight,
+                weight_scale=max_w_scale,
```
Comment (BowenBao) [inline vllm/model_executor/layers/quantization/quark/schemes/quark_w8a8_fp8.py:47]: both `weight` and `max_w_scale` are undefined if `current_platform.is_rocm()` is False.

**Review by BowenBao — CHANGES_REQUESTED** [review]

**Comment by BowenBao** [comment]
Do you observe better accuracy with this change? Mathematically the order should not matter.

File: vllm/model_executor/layers/quantization/quark/schemes/quark_w8a8_fp8.py (hunk 1 above)
Comment (fxmarty-amd) [reply vllm/model_executor/layers/quantization/quark/schemes/quark_w8a8_fp8.py:47]: Fixed, thanks!

**Review by BowenBao — APPROVED** [review]

**Comment by BowenBao** [comment]
it would be even better to update `requantize_with_max_scale` to infer the right dtype to use from weight tensor instead of hardcoding on fp8e4m3_fn.
